app/services/subscriptions/billing_service.rb

```diff
@@ -164,6 +190,47 @@ def monthly_anniversary
       )
     end
 
+    # NOTE: Billed quarterly on anniversary date
+    # rubocop:disable Layout/LineLength
+    def quarterly_anniversary
+      billing_day = <<-SQL
+        DATE_PART('day', (subscriptions.subscription_at#{at_time_zone})) = ANY (
+          -- Check if today is the last day of the month
+          CASE WHEN DATE_PART('day', (#{end_of_month})) = DATE_PART('day', :today#{at_time_zone})
+          THEN
+            -- If so and if it counts less than 31 days, we need to take all days up to 31 into account
+            (SELECT ARRAY(SELECT generate_series(DATE_PART('day', :today#{at_time_zone})::integer, 31)))
+          ELSE
+            -- Otherwise, we just need the current day
+            (SELECT ARRAY[DATE_PART('day', :today#{at_time_zone})])
+          END
+        )
+      SQL
+
+      billing_month = <<-SQL
+        (
+          -- We need to avoid zero and instead of it use 12. E.g.: (3 + 9) % 12 = 0 -> 12
+          CASE WHEN MOD(CAST(DATE_PART('month', (subscriptions.subscription_at#{at_time_zone})) AS INTEGER), 3) = 0
+          THEN
+            (DATE_PART('month', :today#{at_time_zone}) IN (SELECT unnest(ARRAY[3, 6, 9, 12])))
```

Comment:
Not sure about the SELECT UNNEST, the `IN` operator with a list is not enough?
```suggestion
            (DATE_PART('month', :today#{at_time_zone}) IN (3, 6, 9, 12)
```

Reply:
It hasn't worked without unnest but there is a chance that I was using wrong brackets 😅  Working fine with new code 👍